enhancements/insights/config-api.md

```diff
@@ -0,0 +1,245 @@
+  and it does not make sense to expose everything in the new configuration API 
+- Removing the original configuration via `support` secret
+
+## Proposal
+
+The proposal is to introduce a new Insights `CustomResourceDefinition` in the OpenShift API providing a way to configure the Insights operator and to report its status. The new CRD should follow a common CRD structure providing `spec` and `status` attributes. 
+
+```go
+type InsightsSpec struct {
+	// Gathering spec attribute includes all the configuration options related to
+	// gathering of the Insights archive and its uploading to the ingress.
+	GatheringConfig GatheringConfig `json:"gatheringConfig,omitempty"`
+}
+
+type InsightsStatus struct {
+	// GatheringStatus provides basic information about the last Insights gathering.
+	GatheringStatus GatheringStatus `json:"gatheringStatus,omitempty"`
+	// SimpleContentAccessStatus provides basic information about the SCA controller.
+	SimpleContentAccessStatus SimpleContentAccessStatus `json:"simpleContentAccessStatus,omitempty"`
+	// ClusterTransferStatus provides basic information about the ClusterTransfer controller.
+	ClusterTransferStatus ClusterTransferStatus `json:"clusterTransferStatus,omitempty"`
+	// ReportStatus provides general Insights analysis results.
+	ReportStatus ReportStatus `json:"reportStatus,omitempty"`
+}
+
+type GatheringConfig struct {
+	// EnableObfuscation is flag enabling the global obfuscation of the IP addresses and base domain
+	// in the Insights archive data.
+	EnableObfuscation bool `json:"enableObfuscation,omitempty"`
+	// Enables user to force Insights data gathering by setting a new reason. Gathering can be forced even
+	// when all the gatherers are disabled.
+	ForceGatherReason string `json:"forceGatherReason,omitempty"`
+	// List of gatherers to be excluded from the gathering. All the gatherers can disabled by providing "all" value.
+	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	DisabledGatherers []string `json:"disabledGatherers,omitempty"`
+}
+
+type GatheringStatus struct {
+	// LastGatheringTime is the last time when Insights gathering finished.
+	LastGatheringTime time.Time `json:"lastGatheringTime,omitempty"`
+	// Reason provides last known reason of gathering. This is helpful
+	// especially when gathering was forced by user
+	Reason string `json:"reason,omitempty"`
+	// Status is waiting or running. If it is running then it means there is some gathering in progress.
+	// If it is waiting then a new gathering can be triggered immediately.
+	Status string `json:"status,omitempty"`
+	// Errors is a list of gathering errors from the last gathering.
+	Errors []string `json:"errors,omitempty"`
+	// List of active gatherers in the last gathering
+	Gatherers []string `json:"gatherers,omitempty"`
+}
+
+type ReportStatus struct {
+	// Number of active Insights healthchecks with low severity
+	LowCount int `json:"low,omitempty"`
```

Comment:
#1036 is planning on having a bunch more than severity counts in metrics/Prometheus.  I don't see why we'd want to be less rich in the Kube resource representation.  Can the report status be a gather/report timestamp and a slice of recommendations or something?  Makes life easier for folks with a Kube client but no convenient Prom client.

Reply:
Yes this is good point. I agree that this can be extended (hopefully in next API version?). I am just not sure I want do it right now in this enhancement. The webconsole Insights UI widget is the consumer of these numbers so this is basically the first step to keep the current behavior (the UI widget can swtich to this once it's implemented).

Gathering timestamp is available in the `LastGatheringTime` and recommendations can be added, but I would suggest to do it as a second (or next) step.

Reply:
> hopefully in next API version

Sure, no problems with this while you're tech-preview.

> Gathering timestamp is available in the `LastGatheringTime` and recommendations can be added, but I would suggest to do it as a second (or next) step.

This assumes that the current report data is based on the most recent gather data.  That seems a bit racy, but the race window might be small.  And do reports also include Telemetry-sourced analysis?

Reply:
The latest report is based on the last known archive. If the upload fails, there is no polling or downloading of the report. So not that racy IMO. 
>  And do reports also include Telemetry-sourced analysis?

No. As far as I know we don't connect to telemetry while running the Insights analysis. You know, there are some metrics,alerts included in the archive.....